assets/src/loader.rs

```diff
@@ -0,0 +1,338 @@
+use std::any::{Any, TypeId};
+use std::borrow::Borrow;
+use std::cell::UnsafeCell;
+use std::hash::Hash;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use fnv::FnvHashMap;
+use futures::{Future, Poll};
+use rayon::ThreadPool;
+use specs::{Component, DenseVecStorage};
+
+use asset::AssetSpec;
+
+use store::AnyStore;
+use {Allocator, Asset, BoxedErr, Directory, Format, AssetError, LoadError, Store};
+
+/// Represents a future value of an asset. This future may be
+/// added to the ECS world, where the responsible system can poll it and merge
+/// it into the persistent storage once it is `Ready`.
+pub struct AssetFuture<A, E> {
```

Comment:
So there is something wrong about `AssetFuture`. Calling `Future::wait` on it just blocks forever...